### PR TITLE
[FIX] point_of_sale: correct compute prices of order in OrderManagementScreen

### PR DESCRIPTION
**Pull request by caburj**
When viewing an order in the order management screen that has customer
with fiscal position, the calculation of prices is based on the fiscal
position of the currently active order. This commit fixes the issue
by properly specifying the order in the calculation of taxes based
on the order's customer and/or fiscal position.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by caburj** [comment]
Hi @agr-odoo. Sorry about this but I believe this PR is the proper solution to the problem being solved here: https://github.com/odoo/odoo/pull/69475

In fact, we currently have traceback when the active order is in receipt screen and we try to select an order in the order management screen. https://drive.google.com/file/d/1hPvBSESYaVkaao1UeZy_pEdLE_oARs8E/view

cc @pimodoo

File: addons/point_of_sale/static/src/js/models.js [hunk 1]
```diff
@@ -2076,9 +2076,9 @@ exports.Orderline = Backbone.Model.extend({
         }
         return taxes;
     },
-    _map_tax_fiscal_position: function(tax) {
+    _map_tax_fiscal_position: function(tax, order) {
```
Comment (pimodoo) [inline addons/point_of_sale/static/src/js/models.js:2079]: shouldn't we set order=false by default?

**Comment by agr-odoo** [comment]
> Hi @agr-odoo. Sorry about this but I believe this PR is the proper solution to the problem being solved here: #69475

Indeed I can confirm it is fixing the issue and it's a way better fix, thank you

File: addons/point_of_sale/static/src/js/models.js (hunk 1 above)
Comment (caburj) [reply addons/point_of_sale/static/src/js/models.js]: Yeah that true. I force push the branch for this change.

**Comment by agr-odoo** [comment]
Hello,
we found this to be helpful also for opw-2591421 (and possibly other tickets as well)
May we move it forward? Thanks

**Comment by pimodoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**